src/core/tools.py

```
@@ -309,6 +307,31 @@ def patch_https_url(url: str) -> str:
+                img = img.convert('RGB')  # 移除透明通道
+
+            if grayscale:
+                img = img.convert('L')  # 转换为灰度
+                cv_image = np.array(img)
+            else:
+                cv_image = cv2.cvtColor(np.array(img), cv2.COLOR_RGB2BGR)
+
+            return cv_image
+
+    except Exception as e:
+        raise RuntimeError(f"读取图片失败: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)